### PR TITLE
Increase test coverage: Array.prototype.indexOf

### PR DESCRIPTION
**Pull request by matedabis**
Branch coverage:
Before: 22/26
After: 26/26

JerryScript-DCO-1.0-Signed-off-by: Mate Dabis mdabis@inf.u-szeged.hu

### CODE REVIEW

File: tests/jerry/array-prototype-indexof.js [hunk 1]
```diff
@@ -86,3 +86,36 @@ try {
   assert(e.message === "foo");
   assert(e instanceof ReferenceError);
 }
+
+// Checking behavior when values are unable to find
+try{
```
Comment (rerobika) [inline tests/jerry/array-prototype-indexof.js]: Space before `{`.

File: tests/jerry/array-prototype-indexof.js [hunk 2]
```diff
@@ -86,3 +86,36 @@ try {
...
+  var a = [undefined,undefined,undefined,undefined,undefined];
```
Comment (rerobika) [inline tests/jerry/array-prototype-indexof.js]: Space after `,`.

File: tests/jerry/array-prototype-indexof.js [hunk 3]
```diff
@@ -86,3 +86,36 @@ try {
...
+  for(var i=0; i<a.length; i++)
+    {delete a[i];}
```
Comment (rerobika) [inline tests/jerry/array-prototype-indexof.js]: Spaces around `=, <`, moreover:

```js
for (...) {
  //...
}
```

File: tests/jerry/array-prototype-indexof.js [hunk 4]
```diff
@@ -86,3 +86,36 @@ try {
...
+  a.indexOf("bar");
```
Comment (rerobika) [inline tests/jerry/array-prototype-indexof.js]: `assert (false);`

File: tests/jerry/array-prototype-indexof.js [hunk 5]
```diff
@@ -86,3 +86,36 @@ try {
...
+} catch (e) { }
```
Comment (rerobika) [inline tests/jerry/array-prototype-indexof.js]: `assert (e instanceof TypeError)`

File: tests/jerry/array-prototype-indexof.js [hunk 6]
```diff
@@ -86,3 +86,36 @@ try {
...
+
+// Checking behavior when value is null
+try{
```
Comment (rerobika) [inline tests/jerry/array-prototype-indexof.js]: Ditto.

File: tests/jerry/array-prototype-indexof.js [hunk 7]
```diff
@@ -86,3 +86,36 @@ try {
...
+  assert (false);
+} catch (e) {
+  assert (e instanceof ReferenceError);
+}
+
+// checking behavior when unable to get the first element
+try {
+  var a = [0, 1, 2, 3, 4];
+  Object.defineProperty(a, '0', { 'get' : function () { throw new ReferenceError; } });
+  Array.prototype.indexOf.call(a, "bar")
```
Comment (rerobika) [inline tests/jerry/array-prototype-indexof.js]: `assert (false)`.

File: tests/jerry/array-prototype-indexof.js [hunk 8]
```diff
@@ -86,3 +86,36 @@ try {
...
+
+// Checking behavior when values are unable to find
+try{
+  var a = [undefined,undefined,undefined,undefined,undefined];
+  for(var i=0; i<a.length; i++)
+    {delete a[i];}
+  a.indexOf("bar");
+} catch (e) { }
+
+// Checking behavior when value is null
```
Comment (rerobika) [inline tests/jerry/array-prototype-indexof.js]: // Checking behavior when this value is not coercible to object.

File: tests/jerry/array-prototype-indexof.js [hunk 9]
```diff
@@ -86,3 +86,36 @@ try {
...
+  var o = {};
+  Object.defineProperty(o, 'toString', { 'get' : function () { throw new ReferenceError ("foo"); } });
+  var a = { length : o }
+  Array.prototype.indexOf.call(a, function () { })
+  assert (false);
+} catch (e) {
+  assert (e instanceof ReferenceError);
+}
+
+// checking behavior when unable to get the first element
```
Comment (rerobika) [inline tests/jerry/array-prototype-indexof.js]: checking -> Checking

**Comment by matedabis** [comment]
Corrected

**Comment by matedabis** [comment]
@akosthekiss
The function we are testing: `ecma_builtin_array_prototype_object_splice`
The test file: `/tests/jerry/array-prototype-splice.js`
I dont know yet the result if all test suites are executed. By all test suites of the project do you mean all the tests in `run-tests.py` or all the `.js` files in the `tests` folder, or both?

**Comment by matedabis** [comment]
Closing pull request because the updated version of the coverage tester script wich runs `--jerry-test-suite --test262 --unittests --jerry-tests` tests, shows that the branches of this function are already covered so these changes are not needed.
The script:
https://github.com/matedabis/jerryscript/blob/gcov_coverage_tester/tests/gcov-tests/gcovtester.py

**Pull request closed without merging**